stores/metadata.go

```diff
@@ -1598,14 +1600,17 @@ func (ss *SQLStore) UpdateSlab(ctx context.Context, s object.Slab, contractSet s
 		}
 
 		// loop updated shards
-		for _, shard := range s.Shards {
+		for i, shard := range s.Shards {
 			// ensure the sector exists
 			var sector dbSector
 			if err := tx.
 				Where(dbSector{Root: shard.Root[:]}).
 				Assign(dbSector{
 					DBSlabID:   slab.ID,
-					LatestHost: publicKey(shard.Host)},
+					Idx:        i + 1,
```

Comment:
we're not 0-base indexing because we can use 0 as a sentinel value for corruption?

Reply:
Gorm doesn't set default values. So if we set the index to 0, the column remains NULL which is forbidden by the constraint. So yeah it gives us implicit protection against forgetting to set the index field.